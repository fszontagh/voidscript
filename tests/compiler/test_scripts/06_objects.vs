// Test object operations
object $person = {
    string $name : "John",
    int $age : 30,
    bool $active : true
};

printnl("Person name: ", $person->name);
printnl("Person age: ", $person->age);
printnl("Person active: ", $person->active);

// Modify object properties
$person->name = "Jane";
$person->age = 25;
$person->active = false;

printnl("Updated name: ", $person->name);
printnl("Updated age: ", $person->age);
printnl("Updated active: ", $person->active);

// Object iteration
printnl("Object properties:");
for (string $key, auto $value : $person) {
    printnl("  ", $key, ": ", $value);
}