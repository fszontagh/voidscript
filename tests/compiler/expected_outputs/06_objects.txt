Person name: John
Person age: 30
Person active: true
Updated name: Jane
Updated age: 25
Updated active: false
Object properties:
  name: Jane
  age: 25
  active: false